src/pages/consultations/components/UserRating.tsx

```diff
@@ -13,53 +17,109 @@ import {
   SliderTrack,
   Text,
   Textarea,
+  useToast,
   VStack
 } from '@chakra-ui/react'
 import { useState } from 'react'
 import { FaStar } from 'react-icons/fa'
+import { useParams } from 'react-router-dom'
+import { useRateConsultationMutation, useUpdateRatingConsultationMutation } from '../../../api/hooks/consultationMutationHooks'
+import { KonziError } from '../../../api/model/error.model'
 import { RatingModel } from '../../../api/model/rating.model'
 import { UserModel } from '../../../api/model/user.model'
+import { generateToastParams } from '../../../util/generateToastParams'
+import { ErrorPage } from '../../error/ErrorPage'
 
 type Props = {
   user: UserModel & {
-    ratedByCurrentUser?: boolean
     rating?: RatingModel
   }
+  isParticipant: boolean
+  showRatingButton: boolean
+  refetch: () => {}
 }
 
-export const UserRating = ({ user }: Props) => {
+export const UserRating = ({ isParticipant, user, showRatingButton, refetch }: Props) => {
+  const toast = useToast()
+  const { consultationId } = useParams()
+
+  const { mutate: rateConsultation } = useRateConsultationMutation(
+    () => {
+      toast({ title: 'Sikeresen értékeltél!', status: 'success' })
+      refetch()
+    },
+    (e: KonziError) => {
+      toast(generateToastParams(e))
+    }
+  )
+
+  const { mutate: updateRatingConsultation } = useUpdateRatingConsultationMutation(
+    () => {
+      toast({ title: 'Sikeresen frissítetted az értékelést!', status: 'success' })
+      refetch()
+    },
+    (e: KonziError) => {
+      toast(generateToastParams(e))
+    }
+  )
+
   const [open, setOpen] = useState(false)
-  const [value, setValue] = useState<number>(user?.rating?.value ?? 5)
-  const [text, setText] = useState<string>(user?.rating?.text ?? '')
+  const [value, setValue] = useState<number>(user.rating?.value ?? 5)
+  const [text, setText] = useState<string>(user.rating?.text ?? '')
+  const [anonymous, setAnonymous] = useState<boolean>(user.rating?.anonymous ?? false)
+
+  if (consultationId === undefined || isNaN(+consultationId)) {
```

Comment:
id validation outsource can be used here for example, the function mentioned above

Reply:
https://github.com/kir-dev/konzisite-frontend/issues/80